src/crowd.go

```diff
@@ -1,37 +1,47 @@
 package main
 
 import (
+	"errors"
 	"flag"
 	"fmt"
 	"io/ioutil"
 	"log"
 	"net/http"
 
-	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
-
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/14-bits/crowd/pkg/sink"
 	"github.com/julienschmidt/httprouter"
 )
 
 var port int
 var endpoint string
-var queue string
+var sinkType string
+var sinkUrl string
 
 func init() {
 	flag.IntVar(&port, "port", 2104, "HTTP Server Port")
 	flag.StringVar(&endpoint, "endpoint", "/api/foo", "HTTP path to receive POST requests")
-	flag.StringVar(&queue, "queue", "https://sqs.eu-central-1.amazonaws.com/21042018/foo", "URL for the SQS queue")
+	flag.StringVar(&sinkType, "sink-type", "void", "Sink adapter, e.g.: void, sqs, redis...")
+	flag.StringVar(&sinkUrl, "sink-url", "void", "Connection String/URL for the the sink")
 	flag.Parse()
 }
 
-type Queue struct {
-	Client sqsiface.SQSAPI
-	URL    string
+type Sink interface {
+	Push(payload []byte) error
 }
+
 type Crowd struct {
-	q Queue
+	s Sink
+}
+
+func SinkSelect(sinkType string, sinkUrl string) (Sink, error) {
+	switch sinkType {
+	case "sqs":
```

Comment:
@mauricioabreu is there any better way of doing this? hash map?

Reply:
You can reach the same result with hash maps. I guess your concern here is about elegance, right?
`switch` looks good here.

(using a hash map you would need to declare a map of all available sinks)

Reply:
I might move the names to constants,... just in case.